### PR TITLE
Fix resource paths are not preserved order in new versioned API

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
Fixed https://github.com/wso2/product-apim/issues/4845

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 1]
```diff
@@ -2444,12 +2446,13 @@ public void createNewAPIVersion(API api, String newVersion) throws DuplicateAPIE
             String resourcePath = APIUtil.getOpenAPIDefinitionFilePath(api.getId().getApiName(),
                     api.getId().getVersion(), api.getId().getProviderName());
             if (registry.resourceExists(resourcePath + APIConstants.API_OAS_DEFINITION_RESOURCE_NAME)) {
-                JSONObject swaggerObject = (JSONObject) new JSONParser()
-                        .parse(definitionFromOpenAPISpec.getAPIDefinition(api.getId(), registry));
-                JSONObject infoObject = (JSONObject) swaggerObject.get("info");
+                String apiDefinition = definitionFromOpenAPISpec.getAPIDefinition(api.getId(), registry);
+                LinkedHashMap map = new ObjectMapper().readValue(apiDefinition, LinkedHashMap.class);
```
Comment (TanyaM) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Descriptive name for variable map :)

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 2]
```diff
@@ -2444,12 +2446,13 @@ public void createNewAPIVersion(API api, String newVersion) throws DuplicateAPIE
...
+                Map infoObject = (Map) map.get("info");
                 infoObject.remove("version");
                 infoObject.put("version", newAPI.getId().getVersion());
-                definitionFromOpenAPISpec.saveAPIDefinition(newAPI, swaggerObject.toJSONString(), registry);
+                String json = new ObjectMapper().writeValueAsString(map);
```
Comment (TanyaM) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: same as above

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 1 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Fixed with 32e0ffac6b929366b9b43de63779401cc71b04ec

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 2 above)
Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Fixed with 32e0ffac6b929366b9b43de63779401cc71b04ec

**Pull request merged**